cli/harness/src/main/java/io/helidon/build/cli/harness/CommandParser.java

```diff
@@ -112,6 +118,35 @@ static CommandParser create(String... args) {
         return new CommandParser(argsList, commandName, new Resolver(params, properties), error);
     }
 
+    static String[] preProcessArgs(String[] args) {
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < args.length; i++) {
+            if ("--args-file".equals(args[i]) && (i < args.length - 1)) {
+                String argsFile = args[i + 1];
+                result.addAll(readArgsFile(argsFile));
+                i++;
+            } else {
+                result.add(args[i]);
+            }
+        }
+        return result.toArray(new String[0]);
+    }
+
+    static List<String> readArgsFile(String argsFile) {
+        List<String> result = new ArrayList<>();
+        try {
+            Path filePath = Path.of(argsFile);
```

Comment:
Since `File.lines` returns a `Stream<String>` you can simplify this code, e.g.

```java
try {
    return Files.lines(Path.of(argsFile))
            .filter(line -> line.startsWith("#"))
            .map(line.split("\\s+"))
            .collect(Collectors.toList())
} catch (IOException e) {
    throw new UncheckedIOException(e);
}
```

Reply:
done